src/Packlist/PacklistForm.js

```diff
@@ -0,0 +1,218 @@
+import ListItem from '../common/ListItem'
+import usePacklists from './usePacklists'
+
+PacklistForm.propTypes = {
+  onPacklistSave: PropTypes.func.isRequired,
+}
+
+export default function PacklistForm({ onPacklistSave }) {
+  const { register, handleSubmit, reset, errors } = useForm()
+  const [items, setItems] = useState([])
+  const { packlists } = usePacklists()
+  const itemRef = useRef(null)
+  const onSubmit = (packlist, event) => {
+    event.preventDefault()
+    // for testing...
+    if (event?.target && typeof event?.target.reset === 'function')
+      event.target.reset()
+    onPacklistSave({ name: packlist.name, packlist: items })
+  }
+  const handlePacklistKeyDown = (event) => {
+    if (event.key === 'Enter') {
+      event.preventDefault()
+      addItem({ text: event.target.value, id: uuid() })
+    }
+  }
+
+  return (
+    <>
+      <Form onSubmit={handleSubmit(onSubmit)}>
+        <FormInputContainer>
+          <PacklistNameInputLabel htmlFor="name">
+            Name of the new PackList:
+          </PacklistNameInputLabel>
+          <Input
+            placeholder="PackList name"
+            id="name"
+            name="name"
+            ref={register({
+              required: true,
+              minLength: 3,
+              maxLength: 10,
+              validate: {
+                length: (value) =>
+                  value?.trim().length >= 3 && value?.trim().length <= 10,
+                nameTaken: (value) =>
+                  !packlists.find((packlist) => packlist.name === value),
+              },
+            })}
+          />
+          {errors.name?.type === 'required' && (
+            <ErrorMessage>Name is required!</ErrorMessage>
+          )}
+          {errors.name?.type === 'nameTaken' && (
+            <ErrorMessage>Name is taken!</ErrorMessage>
+          )}
+          {(errors.name?.type === 'validate' ||
+            errors.name?.type === 'minLength') && (
+            <ErrorMessage>
+              This field requires at least 3 characters!
+            </ErrorMessage>
+          )}
+          {(errors.name?.type === 'validate' ||
+            errors.name?.type === 'maxLength') && (
+            <ErrorMessage>
+              The name can reach a maximum of 10 characters!
+            </ErrorMessage>
+          )}
+          <Label htmlFor="itemInput">Create new item or task:</Label>
+          <Input
+            placeholder="item you need or task you have to do"
+            id="itemInput"
+            name="item"
+            onKeyDown={handlePacklistKeyDown}
+            ref={(el) => {
+              itemRef.current = el
+              // register(el, {
+              //   required: true,
+              //   minLength: 3,
+              //   maxLength: 20,
+              //   validate: (value) =>
+              //     value &&
+              //     value.trim().length >= 3 &&
+              //     value.trim().length <= 20,
+              // })
```

Comment:
are these comments important or could be deleted?

Reply:
I'm not sure, if the requirements are working without this, this should be tested.. at the first view, it seems like it's working, but I'm not sure